cspreports/models.py

```diff
@@ -120,29 +119,20 @@ def from_message(cls, message):
             # Message is not a valid CSP report. Return as invalid.
             return self
 
-        # Extract individual fields
-        for report_name, field_name in REQUIRED_FIELD_MAP + OPTIONAL_FIELD_MAP:
-            setattr(self, field_name, report_data.get(report_name))
-        # Extract integer fields
-        for report_name, field_name in INTEGER_FIELD_MAP:
-            value = report_data.get(report_name)
-            field = self._meta.get_field(field_name)
-            min_value, max_value = connection.ops.integer_field_range(field.get_internal_type())
-            if min_value is None:
-                min_value = 0
-            # All these fields are possitive. Value can't be negative.
-            min_value = max(min_value, 0)
-            if value is not None and min_value <= value and (max_value is None or value <= max_value):
-                setattr(self, field_name, value)
-        # Extract disposition
-        disposition = report_data.get('disposition')
-        if disposition in dict(DISPOSITIONS).keys():
-            self.disposition = disposition
-
-        # Check if report is valid
+        fields = REQUIRED_FIELDS + OPTIONAL_FIELDS
+        for json_field_name, django_field_name in fields:
+            converter = cls._meta.get_field(django_field_name).to_python
```

Comment:
I was looking into Django fields and I think we should use `clean` instead of `to_python`. It will not only convert the value, but also check if it validates. That should solve the potential problem with storing invalid values to database.

Reply:
@ziima this seems to be a considerable chunk of work that might take some refactoring of the code. I understand that this is already a refactoring ticket, but the original fix was easy and fast to implement and it would help me solve the original issue I had.

Can we consider to move this bit of refactoring to a separate ticket?

Reply:
I think, we only need

```suggestion
            converter = cls._meta.get_field(django_field_name).clean
```

or is there some major issue, I miss?

Reply:
`clean` method on the model instance will validate the model instance as a whole. We might need to use the `clean_fields` method passing the exclude list in case we want to validate a single field, but in case we want to call the clean method then we need to validate the instance as a whole.

I quickly tried to replace the method as you suggested (adding also the instance to validate against, but it broke several tests